AraCorrelator/RayTraceCorrelator.cxx

```diff
@@ -279,6 +228,80 @@ std::map< int, std::vector<int> > RayTraceCorrelator::SetupPairs(
     return pairs;
 }
 
+std::pair< 
+    std::vector< std::vector< std::vector< int > > >,
+    std::vector< std::vector< std::vector< double > > > > RayTraceCorrelator::GetArrivalDelays(std::map<int, std::vector<int > > pairs){
+
+    std::vector< std::vector< std::vector< double > > > delays;
+    std::vector< std::vector< std::vector< int > > > bins;
+
+    for(int solNum=0; solNum<2; solNum++){
+
+        std::vector< std::vector<double> > delays_this_sol;
+        std::vector< std::vector<int> > bins_this_sol;
+
+        for(auto iter = pairs.begin(); iter != pairs.end(); ++iter){
+            int pairNum = iter->first;
+            int ant1 = iter->second[0];
+            int ant2 = iter->second[1];
+
+            std::vector<double> this_delays;
+            std::vector<int> this_global_bins;
+
+            for(int phiBin = dummyMap->GetXaxis()->GetFirst(); phiBin <= dummyMap->GetXaxis()->GetLast(); phiBin++){
+                for(int thetaBin = dummyMap->GetYaxis()->GetFirst(); thetaBin <= dummyMap->GetYaxis()->GetLast(); thetaBin++){
+
+                    int globalBin  = dummyMap->GetBin(phiBin, thetaBin);
+                    double arrival_time1 = arrvialTimes_.at(solNum).at(ant1).GetBinContent(globalBin);
+                    double arrival_time2 = arrvialTimes_.at(solNum).at(ant2).GetBinContent(globalBin);
+                    double dt = arrival_time1 - arrival_time2;
+
+                    // sanity check
+                    if (arrival_time1 < -100 || arrival_time2 < -100) {
```

Comment:
So if it's a nonsense arrival time it will only be a large negative value? Not ever a large positive one?

Reply:
So, the way I coded the arrival time development, it returns large negative numbers if it had no solution. I've never thought about how to check for large positive numbers, since a vertex can in theory be very far away, or maybe the ice is weird and turns out to have a very slow propagation time. So I'm not sure how to check this with generality.

Reply:
No that's fine, make sense.